### ISSUE TITLE
Potential regression for dict get call with fallback

### ISSUE THREAD

**Issue by BrandonNav** [microsoft/pyright#5165]
**Describe the bug**
_I am unsure if this is a regression or new desired inference behaviour. I think it's probably a bug._
When using `dict.get` to retrieve an item from a dictionary with a fallback of another `dict.get`, pyright 1.1.310 infers the return type as the get callable rather than the type of the result.

**To Reproduce**
```python
from typing import reveal_type

my_dict = {
    "item1" : "item1",
    "item2" : "item2",
}

reveal_type(my_dict)
reveal_type(my_dict.get("item1", ""))
reveal_type(my_dict.get("item1", my_dict.get("item2", "")))

```

Pyright 1.1.309
```bash
/private/tmp/test.py
  /private/tmp/test.py:9:13 - information: Type of "my_dict" is "dict[str, str]"
  /private/tmp/test.py:10:13 - information: Type of "my_dict.get("item1", "")" is "str"
  /private/tmp/test.py:11:13 - information: Type of "my_dict.get("item1", my_dict.get("item2", ""))" is "str"
0 errors, 0 warnings, 3 informations
```

Pyright 1.1.310
```bash
/private/tmp/test.py
  /private/tmp/test.py:9:13 - information: Type of "my_dict" is "dict[str, str]"
  /private/tmp/test.py:10:13 - information: Type of "my_dict.get("item1", "")" is "str"
  /private/tmp/test.py:11:13 - information: Type of "my_dict.get("item1", my_dict.get("item2", ""))" is "str | _T@get"
0 errors, 0 warnings, 3 informations
```

**Expected behavior**
The addition of the `_T@get` is a surprise to me. I would except pyright to infer the type of the value from the dict unioned with the fallback value - which I think is the behaviour of 1.1.309

**VS Code extension or command-line**
CLI: 1.1.310

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.312, which I just published. It will also be included in the next insiders release of pylance.
